tests/errmsgs/tproc_mismatch.nim

```diff
@@ -1,12 +1,39 @@
 discard """
   action: reject
-  matrix: "--declaredLocs --hints:off"
+  cmd: '''nim check --hints:off $file'''
   nimoutFull: true
-  nimout: '''tproc_mismatch.nim(12, 49) Error: type mismatch:
- got <proc (a: int, c: float){.cdecl, noSideEffect, gcsafe, locks: 0.}> [proc]
- but expected 'proc (a: int, c: float){.closure, noSideEffect.}' [proc]
-Calling convention: mismatch got '{.cdecl.}', but expected '{.closure.}'.
+  nimout: '''
+tproc_mismatch.nim(26, 52) Error: type mismatch: got <proc (a: int, c: float){.cdecl, noSideEffect, gcsafe, locks: 0.}> but expected 'proc (a: int, c: float){.closure, noSideEffect.}'
+  Calling convention: mismatch got '{.cdecl.}', but expected '{.closure.}'.
+tproc_mismatch.nim(30, 6) Error: type mismatch: got <proc (){.inline, noSideEffect, gcsafe, locks: 0.}>
+but expected one of: 
+proc bar(a: proc ())
+  first type mismatch at position: 1
+  required type for a: proc (){.closure.}
+  but expression 'fn1' is of type: proc (){.inline, noSideEffect, gcsafe, locks: 0.}
+  Calling convention: mismatch got '{.inline.}', but expected '{.closure.}'.
+
+expression: bar(fn1)
+tproc_mismatch.nim(34, 8) Error: type mismatch: got <proc (){.inline, noSideEffect, gcsafe, locks: 0.}> but expected 'proc (){.closure.}'
+  Calling convention: mismatch got '{.inline.}', but expected '{.closure.}'.
+tproc_mismatch.nim(39, 8) Error: type mismatch: got <proc (){.locks: 0.}> but expected 'proc (){.closure, noSideEffect.}'
+  Calling convention: mismatch got '{.nimcall.}', but expected '{.closure.}'.
+  Pragma mismatch: got '{..}', but expected '{.noSideEffect.}'.
 '''
 """
-func a(a: int, c: float){.cdecl.} = discard
-var b: proc(a: int, c: float){.noSideEffect.} = a
+block: # CallConv/Pragma mismatch
+  func a(a: int, c: float) {.cdecl.} = discard
+  var b: proc(a: int, c: float) {.noSideEffect.} = a
```

Comment:
```nim
when defined case12:
  type Foo = ref object
    x0: int
  var g0 = Foo(x0: 1)
  proc fn1(a: int) = g0.x0 = a
  var fn2: proc(a: int)
  var fn3: proc(a: int) {.gcsafe.}
  fn2 = fn1
  fn3 = fn1
```

gives
```
Error: type mismatch:
 got 'proc (a: int){.locks: 0.}' for 'fn1' [proc]
 but expected 'proc (a: int){.closure, gcsafe.}' [proc]
  Calling convention: mismatch got '{.nimcall.}', but expected '{.closure.}'.
  Pragma mismatch: got '{..}', but expected '{.gcsafe.}'.
    fn3 = fn1
```

is `Calling convention: mismatch` relevant here? because `fn2 = fn1` works